src/Redux/todo-reducer.ts

```diff
@@ -134,61 +173,91 @@ export const todoActions = {
+    } catch (error: any) {
+        yield put(todoActions.getTodosError(error.message))
     }
 }
-export const removeTask = (id: string): ThunkType<TodoActionsType> => {
-    return async (dispatch) => {
-        const response = await todoApi.removeTodo(id)
-        dispatch(todoActions.removeTask(id))
+
+function* fetchToggleProgressTodo(action: {type: string, payload: TodoListItemType}) {
+    try {
+        yield call(todoApi.toggleProgress, action.payload.id, action.payload.progress)      
+        yield put(todoActions.toggleProgressSuccess(action.payload.id))       
+    } catch (e) {
+       console.log(e)
     }
 }
-export const changeTask = (task: TodoListItemType): ThunkType<TodoActionsType> => {
-    return async (dispatch) => {
-        const response = await todoApi.updateTodo(task.id, task.title)
-        dispatch(todoActions.changeTask(task))
+
+function* fetchChangeTodo(action: {type: string, payload: TodoListItemType}) {
+    try {
+        yield call(todoApi.updateTodo, action.payload.id, action.payload.title)      
+        yield put(todoActions.changeTaskSuccess(action.payload))          
+    } catch (e) {
+       console.log(e)
     }
 }
-export const toggleProgress = (task: TodoListItemType): ThunkType<TodoActionsType> => {
-    return async (dispatch) => {
-        const response = await todoApi.toggleProgress(task.id, task.progress)
-        dispatch(todoActions.toggleProgress(task.id))
+
+function* fetchRemoveTodo(action: {type: string, payload: string}) {
+    try {
+        yield call(todoApi.removeTodo, action.payload)      
+        yield put(todoActions.removeTaskSuccess(action.payload))         
+    } catch (e) {
+       console.log(e)
     }
 }
-export const toggleProgressAll = (): ThunkType<TodoActionsType> => {
-    return async (dispatch, getState) => {
-        const progress = getState().todo.progressAll
-        const response = await todoApi.toggleProgressAll(!progress)
-        dispatch(todoActions.toggleProgressAll())
+
+function* fetchAddTodo(action: {type: string, payload: string}) {
+    try {
+        const response: TodoListItemType = yield call(todoApi.addTodo, action.payload)      
+        yield put(todoActions.addTaskSuccess(response))      
+    } catch (e) {
+       console.log(e)
     }
 }
-export const removeTaskAllComplited = (): ThunkType<TodoActionsType> => {
-    return async (dispatch) => {
-        const response = await todoApi.removeCompletedTodos()
-        dispatch(todoActions.removeTaskAllComplited())
+
+function* fetchGetTodo() {
+    try {
+        yield put(todoActions.toggleIsFetching(true))
+        const response: TodoListItemType[] = yield call(todoApi.getTodos)        
+        yield put(todoActions.getTodosSuccess(response.reverse()))
+        yield put(todoActions.toggleIsFetching(false))      
+    } catch (error: any) {
+        yield put(todoActions.getTodosError(error.message))
     }
-}
\ No newline at end of file
+}
```

Comment:
either fetchTodo or getTodo. And it is not fetching one todo, it is fetching multiple todos.